docs/docs/typescript/generated-types.md

```diff
@@ -0,0 +1,157 @@
+---
+description: A look at automatic type generation in Redwood
+---
+
+# Generated Types
+
+To add to the TypeScript (and JavaScript!) experience, Redwood generates types for you.
+These generated types not only include your GraphQL operations, but also your named routes, Cells, scenarios, and tests.
+
+When you run `yarn rw dev`, the CLI watches files for changes and triggers type generation automatically, but you can trigger it manually too:
+
+```
+yarn rw g types
+```
+
+:::tip Getting errors trying to generate types?
+
+If you're getting errors trying to generate types, it's worth checking the GraphQL operations in your Cells and SDLs.
+Make sure that they're syntactically valid, and that every query and mutation on the web side is defined in an `*.sdl.js` file on the api side.
+
+:::
+
+If you're curious, you can find the generated types in the `.redwood/types`, `web/types/graphql.d.ts`, and `api/types/graphql.d.ts` directories. Broadly speaking, Redwood generates the following types:
+
+1. ["mirror" types](https://www.typescriptlang.org/docs/handbook/module-resolution.html#virtual-directories-with-rootdirs) for your components, Cells, and layouts on the web side
+2. types based on your queries and mutations on the web side (in `web/types/graphql.d.ts`)
+3. types for resolvers based on your SDLs on the api side (in `api/types/graphql.d.ts`)
+4. types for testing, `currentUser`, etc.
```

Comment:
I've seen all three used and I'm a little confused if they're different: is it `currentUser`, `CurrentUser`, or CurrentUser / current user (i.e. not a code thing at all?)

Reply:
I.e. since `CurrentUser` is capitalized, is it a type? Whereas `currentUser` is a variable somewhere? And then current user is just the concept?

Reply:
They're interchangable really. But `currentUser` refers to the object/property usually, `CurrentUser` refers to the type.

But it gets tricky, when you say "you can find this in your CurrentUser" - where it means both